pkgs/swift2objc/lib/src/ast/visitor.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:logging/logging.dart';
+
+import '../ast/_core/interfaces/compound_declaration.dart';
+import '../ast/_core/interfaces/declaration.dart';
+import '../ast/_core/interfaces/enum_declaration.dart';
+import '../ast/_core/interfaces/function_declaration.dart';
+import '../ast/_core/interfaces/variable_declaration.dart';
+import '../ast/_core/shared/referred_type.dart';
+import '../ast/ast_node.dart';
+import '../ast/declarations/built_in/built_in_declaration.dart';
+import '../ast/declarations/compounds/class_declaration.dart';
+import '../ast/declarations/compounds/members/initializer_declaration.dart';
+import '../ast/declarations/compounds/members/method_declaration.dart';
+import '../ast/declarations/compounds/members/property_declaration.dart';
+import '../ast/declarations/compounds/protocol_declaration.dart';
+import '../ast/declarations/compounds/struct_declaration.dart';
+import '../ast/declarations/enums/associated_value_enum_declaration.dart';
+import '../ast/declarations/enums/normal_enum_declaration.dart';
+import '../ast/declarations/enums/raw_value_enum_declaration.dart';
+import '../ast/declarations/globals/globals.dart';
+
+final _logger = Logger('swift2objc.visitor');
+
+/// Wrapper around [Visitation] to be used by callers.
```

Comment:
The documentation could be improved.

If I understand it correctly, the visitor does the traversal, and the visitation contains the methods on what to modify for each node type.

"Wrapper" doesn't say what it's responsibility is and isn't.

I'd maybe call the the class that determines the traversal order the `Traverser`. Or `TraversalStrategy` (terminology taken from https://strategoxt.org/, which also explicitly separates the function applied to a node on visit, and the traversal strategy of traversing the whole tree).

Reply:
Done. I prefer to have the names of tightly coupled classes like this be closely related though. `Visitor` and `Visitation` makes them sound more similar than `Traverser` and `Visitation`.